### DISCUSSION TITLE
cgs on real hardware

### DISCUSSION THREAD

**Discussion by sytzemeijer** [CosmosOS/Cosmos#2470] [General]
Does cgs also work on real hardware. And if it doesn't work what can I use.

**Comment by raphm4044** [comment]
Only VBECanvas is working on real hardware

**Comment by raphm4044** [comment]
and VGACanvas

**Comment by sytzemeijer** [comment]
where can i find its documentation?

**Comment by raphm4044** [comment]
it's a normal canvas

**Comment by raphm4044** [comment]
so you can use it like FullScreenCanvas.GetFullscreenCanvas()

**Comment by sytzemeijer** [comment]
@RaphMar2021 that was cgs

**Comment by raphm4044** [comment]
yes, VBECanvas & VGACanvas is a part of CGS

**Comment by sytzemeijer** [comment]
But what's the difference then?

**Comment by raphm4044** [comment]
nothing, just the driver behind it changes

**Comment by sytzemeijer** [comment]
And how should I use them.

**Comment by raphm4044** [comment]
like a normal canvas

**Comment by sytzemeijer** [comment]
how does cosmos know which one I want to use

**Comment by raphm4044** [comment]
idk

**Comment by raphm4044** [comment]
ask a coredev on discord

**Comment by sytzemeijer** [comment]
Thanks

**Comment by sytzemeijer** [comment]
[https://cosmosos.github.io/api/Cosmos.System.Graphics.VGACanvas.html](https://cosmosos.github.io/api/Cosmos.System.Graphics.VGACanvas.html) Here the operation of VGACanvas is explained.You can use `public VGACanvas()` to set the VGACanvas mode

```
The FullScreenCanvas provides two methods to get a canvas instance for the screen.   
It automatically chooses the best available driver to use.  

public static Canvas GetFullScreenCanvas(Mode mode):  
 gets the instance of Canvas representing the complete screen in the specified mode  

public static Canvas GetFullScreenCanvas():   
gets the instance of Canvas representing the complete screen in the best driver available on your platform  

```
